sql/src/test/java/io/crate/analyze/AlterTableAddColumnAnalyzerTest.java

```diff
@@ -199,6 +199,26 @@ public void testAddNewNestedColumnToObjectArray() throws Exception {
     }
 
     @Test
+    public void testAddColumnToObjectTypeMaintainsObjectPolicy() throws Exception {
+        AddColumnAnalyzedStatement analysis = e.analyze(
+            "alter table users add column address['street'] string");
+        List<AnalyzedColumnDefinition> columns = analysis.analyzedTableElements().columns();
+        assertThat(columns.size(), is(2));
+
+        AnalyzedColumnDefinition address = columns.get(0);
+        assertThat(address.ident(), is(ColumnIdent.fromPath("address")));
+        assertThat(address.dataType(), is("object"));
+        assertThat(address.isParentColumn(), is(true));
+        assertThat(address.children().size(), is(1));
+        assertThat(address.getObjectType(), is ("strict"));
+
+        AnalyzedColumnDefinition street = address.children().get(0);
+        assertThat(street.ident(), is(ColumnIdent.fromPath("address.street")));
```

Comment:
I don't think we had a test to make sure the child column is actually added under `strict` conditions. I think we should have one - but I'm happy to move it into a separate test if you prefer that